code/go/0chain.net/chaincore/block/entity.go

```diff
@@ -907,9 +907,9 @@ func (b *Block) ComputeState(ctx context.Context, c Chainer) error {
 	}
 
 	if len(b.Events) > 0 && c.GetEventDb() != nil {
-		go func() {
-			c.GetEventDb().AddEvents(ctx, b.Events)
-		}()
+		go func(events []event.Event) {
+			c.GetEventDb().AddEvents(ctx, events)
+		}(b.Events)
 		b.Events = nil
```

Comment:
@kushthedude that's where it's getting nil :)

Reply:
yeah, the next line it is assigned to nil, nice data race, good catch @ssardana08!